hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -185,21 +185,26 @@ private void assignInserts(WorkloadProfile profile, HoodieEngineContext context)
 
         // first try packing this into one of the smallFiles
         for (SmallFile smallFile : smallFiles) {
-          long recordsToAppend = Math.min((config.getParquetMaxFileSize() - smallFile.sizeBytes) / averageRecordSize,
```

Comment:
IIUC `recordsToAppend` will become negative in the current code when `totalUnassignedInserts` is negative?  and we won't get into any assignment, even if we loop thru all small files?